src/Datasource/EntityTrait.php

```diff
@@ -281,8 +299,10 @@ public function &get($property)
 
         if ($this->_methodExists($method)) {
             $result = $this->{$method}($value);
-            return $result;
+            return $recursor($result, $nested);
```

Comment:
That breaks the return-by-reference required by array-style reference. (Trust me, phpunit lit up like a :christmas_tree: when I tried to simplify further.)